### ISSUE TITLE
Track doesn't play, freezes UI

### ISSUE THREAD

**Issue by jacksongoode** [melgu/TidalSwift#15]
I'm able to sign in and browse but playback never starts on latest version.
macOS 13.6 M1 Pro

**Comment by melgu** [comment]
Hi!

The UI freezes because the app synchronously tries every song in the playlist and I made the stupid decision back then to do that on the main thread 😅

Do you have a paid account? And if yes, which kind?
I don’t use Tidal myself anymore, so I don’t have a paid account to check.

**Comment by jacksongoode** [comment]
Would you like to use my credentials? I'd be happy to support the development in some way!

**Comment by melgu** [comment]
If you would be willing to share your credentials, that would really help out. You can contact me via email (mail@melvin-gundlach.de) and I’ll check what’s going on.

**Comment by melgu** [comment]
I just checked with your login credentials and the latest version of TidalSwift (0.1-49) and I could play back any song. So maybe you need to update the app. You can get the latest version [here](https://github.com/melgu/TidalSwift/releases). Or maybe you need to log out and log back in.

**Comment by jacksongoode** [comment]
Huh, that's curious because it just stalls on selection without showing the song duration. Did you authenticate via token or web?

**Comment by melgu** [comment]
I did authenticate via the web. I added a song and a video to your favorites which definitely worked on my end.

**Comment by jacksongoode** [comment]
Yeah, liking tracks also works on my side as well. Maybe there is some issue with the default playback device? I have an M1 (13.6). Could it be that the hardware is different?

**Comment by melgu** [comment]
I'm on an M1 Mac mini running macOS 14.
If you have Xcode installed, you could run clone this repository and run the project directly from Xcode so you can see the console output. Maybe there are some hints there.

What happens when you try to play the song or the video in your favorites?

**Comment by melgu** [comment]
Please retry with the current version of the app.
